channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -319,7 +319,13 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
         
                     const convertUndefined = _self._senderConfig.convertUndefined() || undefined;
                     // construct an envelope that Application Insights endpoint can understand
-                    let aiEnvelope = Sender.constructEnvelope(telemetryItem, _self._senderConfig.instrumentationKey(), itemCtx.diagLog(), convertUndefined);
+                    // if ikey of telemetry is provided and not empty, envelope will use this iKey instead of senderConfig iKey
+                    let defaultEnvelopeIkey = _self._senderConfig.instrumentationKey();
```

Comment:
This looks like it could be simplified as
```JavaScript
let defaultEnvelopeIkey = telemetryItem.iKey || _self._senderConfig.instrumentationKey()
```